Update CLA documentation (#30793)
ampproject/amphtml: contributing/contributing-code.md

## Contributor License Agreement

AMP requires all contributors to agree to the OpenJSF Contributor License Agreement in order to protect contributors and users in issues of intellectual property.

> To agree to the OpenJSF Contributor License Agreement, visit https://cla-assistant.io/ampproject/amphtml/, read through the agreement, and click "Login with GitHub to Agree."

Reviewer: Turns out we have confusing behavior from the check.

1. For `org/repo` URLs, `https://cla-assistant.io/ampproject/amphtml` (without the trailing slash) works, but `https://cla-assistant.io/ampproject/amphtml/` (with the slash) ends in a 404.
2. For `org`-only URLs, `https://cla-assistant.io/ampproject` (without the trailing slash) ends in a 404, but `https://cla-assistant.io/ampproject/` (with the slash) works.

I'm pretty sure this is a quirk of the CLA Assistant [implementation](https://github.com/cla-assistant/cla-assistant). I'll send out another commit to fix the URLs again.

Meanwhile, can someone confirm that you're seeing this too?

Reviewer 2: This is pretty bad. And I can confirm the behavior you're describing.

Reviewer: I've made the URL explicit by hyperlinking it. [`19000e0` (#30793)](https://github.com/ampproject/amphtml/pull/30793/commits/19000e053d7ae078142743ea383a38b4bb6db65b).

> To agree to the OpenJSF Contributor License Agreement, visit [https://cla-assistant.io/ampproject/amphtml](https://cla-assistant.io/ampproject/amphtml), read through the agreement, and click "Login with GitHub to Agree."

Reviewer: nit: Can we modify this text to exactly match what CLA check says?

![image](https://user-images.githubusercontent.com/26553114/97214841-0c146e00-179a-11eb-86a5-fbcf56a1936b.png)

```suggestion
To agree to the OpenJSF Contributor License Agreement, visit [https://cla-assistant.io/ampproject/amphtml](https://cla-assistant.io/ampproject/amphtml), read through the agreement, and click "Sign in with GitHub to agree."
```

## Contributor License Agreement

AMP requires all contributors to agree to the OpenJSF Contributor License Agreement in order to protect contributors and users in issues of intellectual property.

> To agree to the OpenJSF Contributor License Agreement, visit https://cla-assistant.io/ampproject/amphtml, read through the agreement, and click "Login with GitHub to Agree."

Author: @rsimha is it correct to link to ampproject/amphtml here?  I noticed after I clicked "Agree" there, if I substituted any other ampproject repo it said I had already signed.  Should we just point people to the overall ampproject org?

@tobie in case he knows as well

Reviewer 2: The CLA text at https://cla-assistant.io/ampproject/amphtml isn't the right CLA text.

The right text is the individual CLA listed on https://openjsf.org/about/the-openjs-foundation-cla/ and more specifically this one: https://openjsf.org/wp-content/uploads/sites/84/2020/10/OpenJS-Foundation-ICLA-2020-09-23.pdf

/cc @rsimha

Reviewer 2: Once this is fixed, though, you should be able to give just https://cla-assistant.io/ampproject/ as the URL.

Reviewer: > Should we just point people to the overall ampproject org?

@mrjoro Yes, I think we should, because the CLA check was deployed for the organization and not separately for each repo.

> The CLA text at cla-assistant.io/ampproject/amphtml isn't the right CLA text.

@tobie, can you clarify further? The text that is currently in use was recommended by https://github.com/ampproject/meta/issues/60#issuecomment-681111001, which links to [this](https://openjsf.org/wp-content/uploads/sites/84/2019/11/OpenJS-Foundation-Individual-CLA-2019-11-15.pdf) document. Has the preferred OpenJS CLA text has changed recently? Either way, we certainly can change the text, but it will require everyone to re-sign the CLA.

Reviewer 2: Yeah, getting the new CLA text approved was why this whole process took so long. I can't believe I mistakenly linked you to the previous one. It seems the new one hadn't yet been released at that point, and I didn't notice this when linking it.

Reviewer: @mrjoro @tobie I just tried navigating to https://cla-assistant.io/ampproject (just the org name) and ended up at a 404 page. I think we need to use the full URL (https://cla-assistant.io/ampproject/amphtml). Can one of you check on this to confirm that I'm correct?

Reviewer 2: @rsimha you need the trailing slash: https://cla-assistant.io/ampproject/ works, https://cla-assistant.io/ampproject doesn't.

Reviewer: In that case, I think we'll need to explicitly hyperlink the text with the trailing slash. Otherwise, Github will drop the slash when it displays the URL (like it did in both your comments on this thread.)

Reviewer 2: Oh, wow I'm not seeing that here. That's so weird.

Reviewer: It could be something in my logged-in Chrome profile that's hiding the trailing slash. It's subtle behavior though!

Reviewer 2: Wow, that's weird:

<img width="796" alt="Screen Shot 2020-10-23 at 21 35 17" src="https://user-images.githubusercontent.com/3392/97046511-c8421e80-1577-11eb-9d23-20cba84afeab.png">

Reviewer 2: ```suggestion
To agree to the OpenJSF Contributor License Agreement, visit https://cla-assistant.io/ampproject/amphtml/, read through the agreement, and click "Login with GitHub to Agree."
```

> To agree to the OpenJSF Contributor License Agreement, visit https://cla-assistant.io/ampproject/amphtml, read through the agreement, and click "Login with GitHub to Agree."

Reviewer 2: ```suggestion
To agree to the OpenJSF Contributor License Agreement, visit https://cla-assistant.io/ampproject/amphtml, read through the agreement, and click "Login with GitHub to Agree."

Alternatively, you also have the option of downloading the Contributor License Agreement from https://individual-cla.openjsf.org, filling it in, signing it, writing in your GitHub handle, and emailing it to operations@openjsf.org. As processing your CLA is done manually this takes much longer and is therefore not recommended.
```

Reviewer: Making this change here SGTM. It will make our digital CLA easier to understand and use.

> To agree to the OpenJSF Contributor License Agreement, visit https://cla-assistant.io/ampproject/amphtml/, read through the agreement, and click "Login with GitHub to Agree."

Reviewer 2: ```suggestion
To agree to the OpenJSF Contributor License Agreement, visit https://cla-assistant.io/ampproject/amphtml/, read through the agreement, and click "Login with GitHub to Agree."

Alternatively, you also have the option of downloading the Contributor License Agreement from https://individual-cla.openjsf.org, filling it in, signing it, writing in your GitHub handle, and emailing it to operations@openjsf.org. As processing your CLA is done manually, this takes much longer and is therefore not recommended.
```

> **Make sure that the email you associate with your CLA is the same email address you associate with your commits (likely via the `user.email` Git config as described on GitHub's [Set up Git](https://help.github.com/articles/set-up-git/) page).**

Author: @rsimha is this still the case?  Or does CLA assistant work based on the GitHub id?

@tobie in case he knows as well

Reviewer 2: It uses your Github id, so you can get rid of this altogether.

Reviewer: Can confirm that the check is based on GitHub ID.

Reviewer: I took the liberty of adding a commit to remove this sentence.
